### PR TITLE
Stacks support (Beta)

### PR DESCRIPTION
**Pull request by brandonc**
## Description

Adds Beta support for Stacks

## Testing plan

Tests are subject to Beta skips. See output below.

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ OAUTH_CLIENT_GITHUB_TOKEN=ghp_xxx go test ./... -run "TestStack" -v
?   	github.com/hashicorp/go-tfe/examples/backing_data	[no test files]
?   	github.com/hashicorp/go-tfe/examples/configuration_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/registry_modules	[no test files]
?   	github.com/hashicorp/go-tfe/examples/run_errors	[no test files]
?   	github.com/hashicorp/go-tfe/examples/state_versions	[no test files]
?   	github.com/hashicorp/go-tfe/examples/users	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]
?   	github.com/hashicorp/go-tfe/mocks	[no test files]
=== RUN   TestStackCreateAndList
=== RUN   TestStackCreateAndList/List_without_options
=== PAUSE TestStackCreateAndList/List_without_options
=== RUN   TestStackCreateAndList/List_with_project_filter
=== PAUSE TestStackCreateAndList/List_with_project_filter
=== RUN   TestStackCreateAndList/List_with_name_filter
=== PAUSE TestStackCreateAndList/List_with_name_filter
=== RUN   TestStackCreateAndList/List_with_sort_options
=== PAUSE TestStackCreateAndList/List_with_sort_options
=== RUN   TestStackCreateAndList/List_with_pagination
=== PAUSE TestStackCreateAndList/List_with_pagination
=== CONT  TestStackCreateAndList/List_without_options
=== CONT  TestStackCreateAndList/List_with_sort_options
=== CONT  TestStackCreateAndList/List_with_name_filter
=== CONT  TestStackCreateAndList/List_with_project_filter
=== CONT  TestStackCreateAndList/List_with_pagination
--- PASS: TestStackCreateAndList (9.00s)
    --- PASS: TestStackCreateAndList/List_with_project_filter (2.04s)
    --- PASS: TestStackCreateAndList/List_with_name_filter (2.32s)
    --- PASS: TestStackCreateAndList/List_without_options (2.41s)
    --- PASS: TestStackCreateAndList/List_with_pagination (2.57s)
    --- PASS: TestStackCreateAndList/List_with_sort_options (3.39s)
=== RUN   TestStackReadUpdateDelete
--- PASS: TestStackReadUpdateDelete (10.94s)
PASS
ok  	github.com/hashicorp/go-tfe	24.318s
```

### CODE REVIEW

File: stack.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	err = req.Do(ctx, stack)
+	if err != nil {
+		return nil, err
+	}
+
+	return stack, nil
+}
+
+func (s stacks) Update(ctx context.Context, stackID string, options StackUpdateOptions) (*Stack, error) {
+	req, err := s.client.NewRequest("PATCH", "stacks/"+stackID, &options)
```
Comment (sebasslash) [inline stack.go:163]: :nail_care:  
```go
u := fmt.Sprintf("stacks/%s", urlQueryEscape(stackID))
```

File: stack_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	stack1, err := client.Stacks.Create(ctx, StackCreateOptions{
+		Name: "aa-test-stack",
+		VCSRepo: &StackVCSRepo{
+			Identifier:   "brandonc/pet-nulls-stack",
+			OAuthTokenID: oauthClient.OAuthTokens[0].ID,
+		},
+		Project: &Project{
+			ID: orgTest.DefaultProject.ID,
+		},
+	})
```
Comment (sebasslash) [inline stack_integration_test.go:37]: Let's refactor this into a test helper `createStack()`

**Review by sebasslash — CHANGES_REQUESTED** [review]
Minor things. Should we also add a changelog entry?

File: stack_integration_test.go (hunk 2 above)
Comment (brandonc) [reply stack_integration_test.go:37]: Eh I think I'll wait on this one

**Review by sebasslash — APPROVED** [review]

**Pull request merged**